src/python/crash_analysis/stack_parsing/stack_symbolizer.py

```diff
@@ -534,10 +576,12 @@ def process_stacktrace(self, unsymbolized_crash_stacktrace):
 
 def filter_binary_path(binary_path):
   """Filters binary path to provide a local copy."""
-  if environment.is_android():
-    # Skip symbolization when running it on bad entries like [stack:XYZ].
-    if not binary_path.startswith('/') or '(deleted)' in binary_path:
-      return ''
+  if environment.is_android() or environment.is_lkl_job():
+    # LKL fuzzer name is not full path.
+    if environment.is_android():
```

Comment:
these nested is_android() checks make the logic a bit difficult to follow.

restructure it like so (factoring out the only common bit into _find_binary_path_in_build()

```python
def _find_binary_path_in_build(binary_path):
  binary_filename = os.path.basename(binary_path)
  build_directory = environment.get_value('BUILD_DIR')
  return utils.find_binary_path(build_directory, binary_path)

def filter_binary_path(binary_path):
if environment.is_lkl_job():
   return _find_binary_path_in_build(binary_filename)

if environment.is_android_job():
  
  if not binary_path.startswith('/') or '(deleted)' in binary_path:
        return ''
  local_binary_path = _find_binary_path_in_build(binary_path)
  ...

```